### PR TITLE
Use common table for '*' wildcard

### PR DESCRIPTION
**Pull request by pdelewski**
This PR is about `usingCommonTable` configuration property for `*` wildcard:

```
'*':
  useCommonTable: true
  target: [ my-clickhouse-data-source ]
```

### CODE REVIEW

File: quesma/table_resolver/rules.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ func makeDefaultWildcard(quesmaConf config.QuesmaConfiguration, pipeline string)
 			case config.ClickhouseTarget:
 				useConnectors = append(useConnectors, &ConnectorDecisionClickhouse{
 					ClickhouseTableName: input.source,
+					IsCommonTable:       quesmaConf.CreateCommonTable,
```
Comment (avelanarius) [inline quesma/table_resolver/rules.go:82]:
```suggestion
					IsCommonTable:       quesmaConf.CreateCommonTableForWildcard,
```

File: quesma/quesma/config/config.go [hunk 2]
```diff
@@ -44,9 +44,10 @@ type QuesmaConfiguration struct {
 	DisableAuth                bool
 	AutodiscoveryEnabled       bool
 
-	EnableIngest      bool // this is computed from the configuration 2.0
-	CreateCommonTable bool
-
+	EnableIngest                 bool // this is computed from the configuration 2.0
+	CreateCommonTable            bool
+	CreateCommonTableForWildcard bool // the meaning of this is to create a common table for wildcard indexes
```
Comment (avelanarius) [inline quesma/quesma/config/config.go:49]:
```suggestion
	UseCommonTableForWildcard bool // the meaning of this is to create a common table for wildcard indexes
```

Such naming maybe would make things clearer?

File: quesma/quesma/config/config_v2.go [hunk 3]
```diff
@@ -552,6 +552,9 @@ func (c *QuesmaNewConfiguration) TranslateToLegacyConfig() QuesmaConfiguration {
 					errAcc = multierror.Append(errAcc, fmt.Errorf("invalid target %s in configuration of %s", target, DefaultWildcardIndexName))
 				}
 			}
+			if defaultConfig.UseCommonTable {
+				conf.CreateCommonTableForWildcard = true
```
Comment (avelanarius) [inline quesma/quesma/config/config_v2.go:556]: Shouldn't this also set both `conf.CreateCommonTable = true` and `conf.CreateCommonTableForWildcard = true`? (the first one determines if common table is created, the second one controls `table_resolver`)

File: quesma/quesma/config/config.go (hunk 2 above)
Comment (pdelewski) [reply quesma/quesma/config/config.go:49]: I agree, but should not we change `CreateCommonTable` as well to be consistent?

File: quesma/quesma/config/config_v2.go (hunk 3 above)
Comment (pdelewski) [reply quesma/quesma/config/config_v2.go:556]: ok, that make sense

Comment (pdelewski) [reply quesma/quesma/config/config_v2.go:556]: However, after consideration I don't think it's needed as there are two separate rules:
- makeDefaultWildcard
- makeCommonTableResolver

Comment (avelanarius) [reply quesma/quesma/config/config_v2.go:556]: What I had in mind was this use of `CreateCommonTable`: https://github.com/QuesmaOrg/quesma/blob/87c168d36135bb99b47c0b77ec1e089aa8f65cb5/quesma/main.go#L105-L108

Comment (pdelewski) [reply quesma/quesma/config/config_v2.go:556]: ok, then you are right

**Review by avelanarius — APPROVED** [review]
LGTM, but my personal taste would be to change the name of `CreateCommonTableForWildcard` to `UseCommonTableForWildcard` (or something similar)

**Comment by pdelewski** [comment]
> LGTM, but my personal taste would be to change the name of `CreateCommonTableForWildcard` to `UseCommonTableForWildcard` (or something similar)

ok, I looked at that one more time and I agree as `CreateCommonTable` is used in both contexts for creation and use and the one for wildcard is used only in one context. I will rename it according to your suggestion.

File: quesma/quesma/config/config.go (hunk 2 above)
Comment (pdelewski) [reply quesma/quesma/config/config.go]: done

**Pull request merged**